### PR TITLE
Base class changed to `UIScrollView`, animate selection, detect tags automatically by delimiter, etc.

### PR DESCRIPTION
**Pull request by krezzoid**
+ code style fixes, configured to work with SwiftLint;
+ animation of tag selection;
+ base class changed from `UIView` to `UIScrollView` and added maxHeight parameter - if tags height more than this value scrolling will appear;
+ detect automatically tags by specified delimiter + bool value to show/hide delimiter;
+ fixed sample application (need to be improved in the future);

### CODE REVIEW

**Comment by ricardopereira** [comment]
@krezzoid Thank you for this incredible contribution! I'll take a look and I'll leave a review shortly.

**Comment by ricardopereira** [comment]
@krezzoid Unfortunately, the test suite is failing:

```
❌  /Users/vagrant/git/WSTagsFieldTests/WSTagsFieldTests.swift:62:53: 'tagViews' is inaccessible due to 'fileprivate' protection level
        XCTAssert(tagsField.tags.count == tagsField.tagViews.count)
```

Could you take a look please?

File: Example/WSTagsFieldExample/ViewController.swift [hunk 1]
```diff
@@ -10,107 +10,82 @@ import UIKit
...
+}
+
+extension ViewController {
+    fileprivate func textFieldEventss() {
+        
         tagsField.onDidAddTag = { _ in
             print("DidAddTag")
         }
-
+
```
Comment (ricardopereira) [inline Example/WSTagsFieldExample/ViewController.swift]: Please remove whitespaces.

**Comment by krezzoid** [comment]
@rafaeljegundo done ;)
btw, please take a look tests code coverage report - here is a lot of missed functions

**Comment by ricardopereira** [comment]
@krezzoid Thanks! Yes, test coverage is pretty low :(

**Comment by ricardopereira** [comment]
> base class changed from `UIView` to `UIScrollView` and added maxHeight parameter - if tags height more than this value scrolling will appear;

@krezzoid `UIScrollView` is nice but I know that some users prefer to see all tags at once by extending the height. Maybe this should be an option: `automatically increase/decrease the intrinsic height`. What are your thoughts about it?

**Comment by krezzoid** [comment]
take a look:
`open var maxHeight: CGFloat = CGFloat.infinity`
this means that view will try to increase self size infinitely - for users who don't want to have scrolling

**Pull request merged**